crates/ruff_linter/src/rules/pycodestyle/rules/logical_lines/whitespace_before_parameters.rs

```diff
@@ -76,7 +76,8 @@ pub(crate) fn whitespace_before_parameters(line: &LogicalLine, context: &mut Log
             let end = token.end() - TextSize::from(1);
             let kind: WhitespaceBeforeParameters = WhitespaceBeforeParameters { bracket: kind };
 
-            let mut diagnostic = OldDiagnostic::new(kind, TextRange::new(start, end));
+            let mut diagnostic =
+                OldDiagnostic::new(kind, TextRange::new(start, end), context.source_file());
```

Comment:
Nit: We could consider to have a `context.report_diagnostic` method that returns the same guard as `checker.report_diagnostic` but I think this is fine too

Reply:
Oh, I haven't thoubht about the context. Yeah, that could be annoying